src/api/app/controllers/webui/users/subscriptions_controller.rb

```diff
@@ -1,29 +1,38 @@
 class Webui::Users::SubscriptionsController < Webui::WebuiController
-  before_action :require_login
+  # TODO: Remove this when we'll refactor kerberos_auth
+  before_action :kerberos_auth
+
+  after_action :verify_authorized
 
   def index
+    @subscriptions_form = subscriptions_form(default_form: params[:default_form])
+    authorize(@subscriptions_form)
+
     @user = User.session!
     @groups_users = @user.groups_users
 
-    @subscriptions_form = subscriptions_form(default_form: params[:default_form])
-
     respond_to do |format|
       format.html
       format.js
     end
   end
 
   def update
+    @subscriptions_form = subscriptions_form
+    authorize(@subscriptions_form)
+
     User.session!.groups_users.each do |gu|
       gu.email = params[gu.group.title] == '1'
       gu.save
     end
 
-    subscriptions_form.update!(params[:subscriptions]) if params[:subscriptions]
-    flash.now[:success] = 'Notifications settings updated'
-  rescue ActiveRecord::RecordInvalid => e
-    flash.now[:error] = "Notifications settings could not be updated due to an error: #{e.message}"
-  ensure
```

Comment:
I've removed the `ensure` catch all error logic because it led into a double render error when `Pundit::NotAuthorizedError` was raised within this action.